### ISSUE TITLE
Replace hardcoded supported ordering with supported ordering variable in upgrade field validatebasic

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#3677]
## Summary

Replace hardcoded ordering support in `UpgradeFields.ValidateBasic()` with a variable to prevent accidental bugs when increasing the list of supported ordering types

## Problem Definition

If we add a new ordering to our supported types, we may forget to update this hardcoded set. We should refer to a variable which defines our global set of supported versions

## Proposal

Replace:
```go
        if !collections.Contains(uf.Ordering, []Order{ORDERED, UNORDERED}) {
		return errorsmod.Wrap(ErrInvalidChannelOrdering, uf.Ordering.String())
	}
```

with something like
```go
        if !collections.Contains(uf.Ordering.String(), connectiontypes.SupportedOrderings) {
		return errorsmod.Wrap(ErrInvalidChannelOrdering, uf.Ordering.String())
	}
```

This will probably require pulling out the supported ordering out of `connectiontypes.DefaultIBCVersion` and placing them into:
```go
var (
    DefaultIBCVersion = NewVersion(DefaultIBCVersionIdentifier, SupportedOrderings)
    
    SupportedOrderings = []string{"ORDER_ORDERED", "ORDER_UNORDERED"}
)
```

I think it makes sense to first:
- [ ] open a pr to main, splitting the supported orderings into their own field
- [ ] when 04-channel-upgrades branch is updated to latest main, make the change on that branch to use the new field

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by crodriguezvega** [comment]
Should we add (in a separate issue/PR) a similar `collections.Contains ...` check in `ValidateBasic` of `MsgChannelOpenInit` and `MsgChannelOpenTry` to check the [`Ordering` of `Channel`](https://github.com/cosmos/ibc-go/blob/main/proto/ibc/core/channel/v1/channel.proto#L19)?

**Comment by damiannolan** [comment]
If we use `connectiontypes.SupportedOrderings` and then a new ordering is added - then the code on chain is saying "yes, this is a supported ordering" but the open connection end stored in state was negotiated under the original connection handshake parameters. So we would need connection upgradability to add a new ordering type, right? 
I'm not sure if its 100% safe/forwards compatible to use `connectiontypes.SupportedOrderings` out of the box. Any thoughts?

**Comment by colin-axner** [comment]
Good point @damiannolan! We should move the check in `upgradeFields.ValidateBasic()` to `k.ValidateUpgradeFields()` and use the proposed connection ID's supported orderings (features) to do the collection check? (also maybe give `validateUpgradeFields` a more descriptive name?)

We do need to use connection upgradability to add new ordering types to an existing connection

**Comment by colin-axner** [comment]
I think it is safe to use the `SupportedOrderings` var for basic validation (stateless checks) since it is essentially just checking that the ordering is conceivably supported (ie the ordering exists)

**Comment by damiannolan** [comment]
That makes sense to me, and provides more defence in depth!
